File: scripts/test_openai.py
import os
import requests
import json

api_key = os.getenv("OPENAI_API_KEY")

if not api_key:
    print("[v0] ERROR: OPENAI_API_KEY no está definida")
    exit(1)

print(f"[v0] OPENAI_API_KEY detectada: {api_key[:10]}...")

url = "https://api.openai.com/v1/chat/completions"
headers = {
    "Content-Type": "application/json",
    "Authorization": f"Bearer {api_key}"
}

payload = {
    "model": "gpt-4o-mini",
    "messages": [
        {
            "role": "user",
            "content": "Test: Eh nene, que tu ta haciendo ahora? (Dominican Spanish)"
        }
    ],
    "temperature": 0.7,
    "max_tokens": 100
}

print("[v0] Enviando request a OpenAI...")
print(f"[v0] URL: {url}")
print(f"[v0] Headers: {headers}")

try:
    response = requests.post(url, json=payload, headers=headers, timeout=30)
    print(f"[v0] Response status: {response.status_code}")
    
    data = response.json()
    
    if "error" in data:
        print(f"[v0] ERROR de OpenAI: {data['error']}")
    else:
        print("[v0] SUCCESS! Respuesta:")
        print(data["choices"][0]["message"]["content"])
except Exception as e:
    print(f"[v0] Exception: {str(e)}")

File: scripts/test-openai.mjs
import fetch from "node-fetch"

const API_KEY = process.env.OPENAI_API_KEY
const BASE_URL = "https://api.openai.com/v1/chat/completions"

if (!API_KEY) {
  console.error("[v0] ERROR: OPENAI_API_KEY no está definida")
  process.exit(1)
}

console.log("[v0] OPENAI_API_KEY detectada:", API_KEY.slice(0, 10) + "...")

const testPayload = {
  model: "gpt-4o-mini",
  messages: [
    {
      role: "user",
      content: "Test message in Dominican Spanish: 'Eh nene, que tu ta haciendo ahora?'",
    },
  ],
  temperature: 0.7,
  max_tokens: 100,
}

console.log("[v0] Enviando request a OpenAI...")
console.log("[v0] Payload:", JSON.stringify(testPayload, null, 2))

fetch(BASE_URL, {
  method: "POST",
  headers: {
    "Content-Type": "application/json",
    Authorization: `Bearer ${API_KEY}`,
  },
  body: JSON.stringify(testPayload),
})
  .then((res) => {
    console.log("[v0] Response status:", res.status)
    return res.json()
  })
  .then((data) => {
    if (data.error) {
      console.error("[v0] OpenAI Error:", data.error)
    } else {
      console.log("[v0] SUCCESS! Respuesta de OpenAI:")
      console.log(data.choices[0].message.content)
    }
  })
  .catch((err) => {
    console.error("[v0] Error:", err.message)
  })
